clash-lib/src/Clash/Normalize/Transformations.hs

```diff
@@ -350,15 +350,17 @@ elemExistentials _ e = return e
 
 -- | Move a Case-decomposition from the subject of a Case-decomposition to the alternatives
 caseCase :: HasCallStack => NormRewrite
-caseCase _ e@(Case (stripTicks -> Case scrut alts1Ty alts1) alts2Ty alts2)
+caseCase (TransformContext is0 _) e@(Case (stripTicks -> Case scrut alts1Ty alts1) alts2Ty alts2)
   = do
     ty1Rep <- representableType <$> Lens.view typeTranslator
                                 <*> Lens.view customReprs
                                 <*> pure False
                                 <*> Lens.view tcCache
                                 <*> pure alts1Ty
     if not ty1Rep
-      then let newAlts = map (second (\altE -> Case altE alts2Ty alts2)) alts1
+      then let newAlts = map
+                           (second (\altE -> Case altE alts2Ty alts2))
+                           (map (deShadowAlt is0) alts1)
```

Comment:
Add a Note comment on why we need to deshadow, like we do in AppProp

Reply:
Done